### PR TITLE
Fixes #4182: Added dark mode support to OptionActivity and OptionWithoutDrawerActivity

### LINKED ISSUE #4182: Add dark mode support to OptionActivity and OptionWithoutDrawerActivity
Add dark mode support to OptionActivity and OptionWithoutDrawerActivity as well as the contained views.

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by Akshatkamboj14**
## Explanation

**Fixes #4182 
Added dark mode support to OptionActivity and OptionWithoutDrawerActivity**

I have introduced two drawable resource files  **_general_item_border.xml_** and **_general_item_border_cyan.xml_** as the original drawable resource files  **_general_item_background_border.xml_** and **_general_item_background_border_cyan.xml_** are getting used in 7 different places and to have a consistent UI throughout the application so separate drawable resource files should be preferable.

| **Default Mode** | **After Dark Mode** |
| ------------- | ------------- |
| image | image |

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Akshatkamboj14** [comment]
@rt4914 @ayush0402 PTAL, thanks.

File: app/src/main/res/layout-sw600dp/option_story_text_size.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     android:id="@+id/reading_text_size_item_layout"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:background="@{commonViewModel.selectedFragmentIndex == viewModel.itemIndex ? @drawable/general_item_background_border_cyan : @drawable/general_item_background_border}"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp/option_story_text_size.xml:20]: I think this condition is important so as it show the divider which is not visible in your screenshots now.

File: app/src/main/res/layout/option_audio_language.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     android:id="@+id/audio_laguage_item_layout"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:background="@{commonViewModel.selectedFragmentIndex == viewModel.itemIndex ? @drawable/general_item_background_border_cyan : @drawable/general_item_background_border}"
+    android:background="@color/component_color_option_activity_options_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout/option_audio_language.xml]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Akshatkamboj14 Suggested changes

**Comment by Akshatkamboj14** [comment]
@rt4914 @ayush0402 @BenHenning **PTAL.**

**Comment by BenHenning** [comment]
@ayush0402 can you please take a first pass? I'm otherwise assuming @rt4914 will follow up as the code owner, but please let me know if I need to review anything here.

File: app/src/main/res/drawable/general_item_border.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_option_activity_options_background_color" />
+  <stroke
+    android:width="1dp"
+    android:color="@color/color_def_black_12" />
+</shape>
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_border.xml:8]: Please mention the in the PR description why this and the other file has been introduced.

**Review by rt4914 — APPROVED** [review]
LGTM, just one change in PR description is required.

File: app/src/main/res/drawable/general_item_border.xml (hunk 3 above)
Comment (Akshatkamboj14) [reply app/src/main/res/drawable/general_item_border.xml:8]: Okay @rt4914

**Comment by BenHenning** [comment]
Hi. Since I was requested as a reviewer on this PR: Please note that I will be away & unavailable for reviews starting today until 23 May 2022. Please try to make any progress that you can which doesn't require me, and I'll catch up on my reviews when I return. Thanks for your flexibility--I really appreciate it.

**Pull request merged**